packages/contracts-bedrock/src/L2/GasPriceOracle.sol

```diff
@@ -84,6 +84,7 @@ contract GasPriceOracle is ISemver {
     /// @param _data Unsigned fully RLP-encoded transaction to get the L1 gas for.
     /// @return Amount of L1 gas used to publish the transaction.
     function getL1GasUsed(bytes memory _data) public view returns (uint256) {
+        _data = flzCompress(_data);
         uint256 total = 0;
         uint256 length = _data.length;
         for (uint256 i = 0; i < length; i++) {
```

Comment:
Don't think you want/need the 0/non-zero counting here since this is for the compression approximation which you are replacing. Just multiply length by 16 to keep units roughly the same?  (Who knows how many 1s/0s it actually ends up with when it hits the batch compressor....)

Reply:
When I ran my analysis I put all the post-compressed data back through the 0/non-0 counting algorithm to get it back into the calldata magnitude. My thinking was that, even though the compression ratio wasn't as efficient, the ratio of 0s/non-0s would be comparable. It would be interesting to compare the msq error between the 0/non-0 vs just using the compressed size.

If we don't do the counting, we'd have to change the scalar to something between 12-15 to account for the actual calldata cost, which is not necessarily a problem but somewhat less understandable than a decimal in the 0-1 range. It also doesn't make sense to do this counting when estimating for 4844, so that's a argument for simplification if the msq is roughly equivalent.

Reply:
Switched to remove the counting as 0s in compressed data are rare so there wasn't any point keeping this